options.go

```diff
@@ -279,15 +279,20 @@ func DefaultStaticRelays() Option {
 	}
 }
 
-// WithReachability overrides automatic reachability detection to force the local node
-// to believe it is either unreachable or reachable externally.
-func WithReachability(reachable bool) Option {
+// WithNATPublicReachability overrides automatic reachability detection to force the local node
+// to believe it is reachable externally.
+func WithNATPublicReachability() Option {
```

Comment:
* NAT isn't _actually_ relevant. AutoNAT is poorly named...
* `With` is also funky given that we're not using that idiom for other libp2p options.
* Given the description, I fell like this should use the word force somewhere.

Maybe:

* `ForceReachabilityPublic`, `ForceReachabilityPrivate`?